### PR TITLE
Add support for arena allocation.

### PR DESCRIPTION
**Pull request by zrlk**
(and keep pointer sigils where they belong)

### CODE REVIEW

File: anodyne/base/arena.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "anodyne/base/arena.h"
+
+namespace anodyne {}
```
Comment (creachadair) [inline anodyne/base/arena.cc]: I'm confused why this is needed. Does the compiler get cranky if we don't have at least one decl?

File: anodyne/base/arena.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+#define ANODYNE_BASE_ARENA_H_
+
+#include <cstddef>
+#include <cstdlib>
+#include <vector>
+
+namespace anodyne {
+
+/// \brief Performs bump-pointer allocation of pointer-aligned memory.
+/// \warning Since `Arena`-allocated objects never have their destructors
```
Comment (creachadair) [inline anodyne/base/arena.h]: Shouldn't there be some way to destroy the Arena itself, and thence its allocated values? I wonder what will happen when we try to wire up a tool built on this as a long-running process.

That may be a larger issue than we want to tackle here, but it stood out.

File: anodyne/base/arena.cc (hunk 1 above)
Comment (zrlk) [reply anodyne/base/arena.cc]: Apparently not. I thought Bazel got upset, but not today.

File: anodyne/base/arena.h (hunk 2 above)
Comment (zrlk) [reply anodyne/base/arena.h]: Yes; the Arena's destructor frees all of its blocks. It just won't, say, destroy a unique_ptr if someone's put one in an ArenaObject.

Comment (creachadair) [reply anodyne/base/arena.h]: Ah, right—it isn't typed so you can't invoke them explicitly.

**Pull request merged**